api/go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/hashicorp/boundary/api
 
 go 1.23.0
```

Comment:
Do you mean instead of adding `toolchain go1.23.1`?

Reply:
It seems to me that `toolchain` is only necessary if we want to specify a different `minimum` supported version and preferred version of Go. If we just update the `go` line to the same as `.go-version`, the `toolchain` directive should be unnecessary. I was reading https://go.dev/doc/modules/gomod-ref#toolchain and https://go.dev/doc/toolchain.

Reply:
I used `toolchain` because I didn't see anything in the [Go 1.23.1 milestone issue tracker](https://github.com/golang/go/issues?q=milestone%3AGo1.23.1+label%3ACherryPickApproved) that I thought directly impacted Boundary. Are there any guidelines on when to use `toolchain` instead of updating the `go` declaration?

Reply:
It's honestly pretty unclear to me exactly what the use case is for `toolchain`, but I think it would be simpler for us just to keep updating the `.go-version` file and `go` directive, and it should have the same effect. I don't see why we would need to support building with different toolchain versions since our CI is running of the version in `.go-version`, which is now `1.23.1`.

Reply:
Sounds good to me. I will update the pull request.